src/test/scala/au/com/agiledigital/toolform/app/ToolFormAppTest.scala

```diff
@@ -17,62 +20,80 @@ class ToolFormAppTest extends FlatSpec with Matchers {
+                         |		public-api ==> 'HTTP Public API'
+                         |		se_swip_elastic-search ==> 'SE Elastic Search'
+                         |		se-swip-influx-db ==> 'SE Influx DB'
+                         |		client/public ==> 'SE Public Web Application'
+                         |	Resources:
+                         |		se-swip-mail-relay
+                         |		se-swip-carbon
+                         |		se-swip-db
+                         |	Links:
+                         |		se_swip_elastic-search -> public-api
+                         |		se-swip-mail-relay -> public-api
+                         |		se-swip-carbon -> public-api
+                         |		se-swip-db -> public-api
+                         |		se-swip-influx-db -> public-api
+                         |""".stripMargin)
   }
 
   "inspect file that does not exist" should "display error string" in {
-    val result = ToolFormApp.execute(List("inspect", "-i", "bad.txt").toArray)
-    result.left.get.message should equal("File [bad.txt] does not exist.")
+    val result = simulateAppForTest(List("inspect", "-i", "bad.txt").toArray)
+    result should startWith("File [bad.txt] does not exist.")
   }
 
   "inspect malformed file" should "display error string" in {
-    val result = ToolFormApp.execute(List("inspect", "-i", malformedFile.getAbsolutePath).toArray)
-    result.left.get.message should include("Failed to parse project configuration")
+    val result = simulateAppForTest(List("inspect", "-i", malformedFile.getAbsolutePath).toArray)
+    result should startWith("Failed to parse project configuration")
   }
 
   "bad argument" should "display error string" in {
-    val result = ToolFormApp.execute(List("--bad", "bad").toArray)
-    result.left.get.message should include("Invalid arguments")
+    val result = simulateAppForTest(List("--bad", "bad").toArray)
+    result should startWith("Unexpected option: --bad")
   }
 
-  "--inspect with missing argument option" should "display error string" in {
-    val result = ToolFormApp.execute(List("inspect").toArray)
-    result.left.get.message should include("Invalid arguments")
+  "inspect with missing argument option" should "display error string" in {
+    val result = simulateAppForTest(List("inspect", "-i").toArray)
+    result should startWith("Missing value for option")
   }
 
   it should "fail if no arguments are specified" in {
-    val result = ToolFormApp.execute(List("").toArray)
-    result.left.get.message should include("Invalid arguments")
+    val result = simulateAppForTest(List("").toArray)
+    result should startWith("Unexpected argument")
   }
 
-  // TODO: Empty components/Resources/Links combos
-  // TODO: Malformed components/Resources/Links combos
-  // TODO: Test substitution
+  "plugin loader" should "load the inspect plugin" in {
+    val plugins = ToolFormPluginLoader.loadPlugins
+    plugins.size should be > 1
+    plugins.head shouldBe a[InspectCommand]
```

Comment:
Be aware that `head` is a partially defined function - it will fail if the seq is empty. Definitely don't want to use it production code.

http://www.scalatest.org/user_guide/using_matchers#workingWithSequences has some examples of matching on the content of collections.

Reply:
I thought I could get away with it in the test here since I assert the size of the seq on the line above.  Agree it is a bit dodgy, but I wasn't sure how else to do the type check in scalatest.